contracts/Marketplace.sol

```diff
@@ -208,35 +290,44 @@ contract Marketplace is Ownable {
     function grantSubscription(bytes32 productId, uint subscriptionSeconds, address recipient) public whenNotHalted onlyProductOwner(productId){
         return _subscribe(productId, subscriptionSeconds, recipient, false);
     }
+
+
     function buyFor(bytes32 productId, uint subscriptionSeconds, address recipient)  public whenNotHalted {
         return _subscribe(productId, subscriptionSeconds, recipient, true);
     }
 
+
      /**
      * Purchases access to this stream for msg.sender.
      * If the address already has a valid subscription, extends the subscription by the given period.
      * @dev since v4.0: Notify the seller if the seller implements PurchaseListener interface
      */
     function buy(bytes32 productId, uint subscriptionSeconds) public whenNotHalted {
-        buyFor(productId,subscriptionSeconds,msg.sender);
+        emit SubscriptionImported(productId, msg.sender,123);
```

Comment:
This row seems accidental

Reply:
oops. I'll remove. That was a primitive debugging aid...